typespec-extension/src/code-model-builder.ts

```diff
@@ -2347,158 +2209,54 @@ export class CodeModelBuilder {
-    //       language: schema.language,
-    //       summary: schema.summary,
-    //       extensions: schema.extensions,
-    //       ...decorators,
-    //     });
-    //   }
-    // } else if (schema instanceof NumberSchema) {
-    //   const decorators = {
-    //     minimum: getMinValue(this.program, prop),
-    //     maximum: getMaxValue(this.program, prop),
-    //   };
-
-    //   if (Object.values(decorators).some((it) => it !== undefined)) {
-    //     schema = new NumberSchema(
-    //       schema.language.default.name,
-    //       schema.language.default.description,
-    //       schema.type,
-    //       schema.precision,
-    //       {
-    //         language: schema.language,
-    //         summary: schema.summary,
-    //         extensions: schema.extensions,
-    //         ...decorators,
-    //       },
-    //     );
-    //   }
-    // }
-    const format = getFormat(this.program, prop);
-    if (format) {
-      // TODO: deprecate format
-      if (prop.type.kind === "Scalar" && schema instanceof StringSchema) {
-        schema = this.processFormatString(prop.type, format, nameHint);
-      }
-    } else {
-      // encode
-      const encode = getEncode(this.program, prop);
-      if (encode) {
-        let type = prop.type;
-        if (prop.type.kind === "Union" && isNullableType(prop.type)) {
-          const nonNullVariants = Array.from(prop.type.variants.values()).filter((it) => !isNullType(it.type));
-          type = nonNullVariants[0].type;
-        }
-        if (type && type.kind === "Scalar") {
-          if (encode.encoding === "seconds" && hasScalarAsBase(type, "duration")) {
-            schema = this.processDurationSchema(type, nameHint, getDurationFormat(encode));
-          } else if (
-            (encode.encoding === "rfc3339" || encode.encoding === "rfc7231" || encode.encoding === "unixTimestamp") &&
-            (hasScalarAsBase(type, "utcDateTime") || hasScalarAsBase(type, "offsetDateTime"))
-          ) {
-            if (encode.encoding === "unixTimestamp") {
-              return this.processUnixTimeSchema(type, nameHint);
-            } else {
-              return this.processDateTimeSchema(type, nameHint, encode.encoding === "rfc7231");
-            }
-          } else if (encode.encoding === "base64url" && hasScalarAsBase(type, "bytes")) {
-            return this.processByteArraySchema(type, nameHint, true);
-          }
-        }
-      }
-    }
-    return schema;
-  }
-
-  private processModelProperty(prop: ModelProperty): Property {
-    const schema = this.processSchema(prop, prop.name);
-    let nullable = isNullableType(prop.type);
+  private processModelPropertyFromSdkType(prop: SdkModelPropertyType): Property {
+    // TODO: question: why the schema name is like this? This case is related with literal.tsp
+    const schemaNameHint = pascalCase(getNamePrefixForProperty(prop.__raw as ModelProperty)) + pascalCase(prop.name);
```

Comment:
Not sure if you still need this `getNamePrefixForProperty`.

Or is it for the optional Constant case?

Reply:
Yes, I first removed it, but found issue in optional literal case. So add it back.